stories/colours.jsx

```diff
@@ -6,23 +6,29 @@ import { storiesOf } from '@storybook/react';
 import { withKnobs, radios } from '@storybook/addon-knobs';
 
 import Colours from '../.storybook/util/colours';
+import { paletteAsMap } from '../scripts/pasteup-sass';
 
-const categories = {
-  brand: ['brand-dark', 'brand-main', 'brand-pastel'],
-  neutrals: ['brightness-7', 'brightness-20', 'brightness-46', 'brightness-60', 'brightness-86', 'brightness-93', 'brightness-97', 'brightness-100'],
-  state: ['highlight-main', 'highlight-dark', 'green-main', 'green-dark', 'error', 'success'],
-  news: ['news-dark', 'news-main', 'news-bright', 'news-pastel', 'news-faded'],
-  opinion: ['opinion-dark', 'opinion-main', 'opinion-bright', 'opinion-pastel', 'opinion-faded'],
-  sport: ['sport-dark', 'sport-main', 'sport-bright', 'sport-pastel', 'sport-faded'],
-  culture: ['culture-dark', 'culture-main', 'culture-bright', 'culture-pastel', 'culture-faded'],
-  lifestyle: ['lifestyle-dark', 'lifestyle-main', 'lifestyle-bright', 'lifestyle-pastel', 'lifestyle-faded'],
-};
-
-const stories = storiesOf('Colour palette', module)
-  .addDecorator(withKnobs);
+/*
+Get the first part of each colour name and make them unique.
+This maps neatly to the exported categories.
+*/
+const palette = paletteAsMap();
+const categories = [...(
+  new Set(Object.keys(palette).map(key => key.split('-')[0]))
```

Comment:
I suggest a simple substring
```js
key.substr(0, key.indexOf('-'))
```

Reply:
that would come off empty if there's no dashes right?

Reply:
Correct, it's just a small detail to avoid array allocation.
```js
key.includes('-') ? ... : key
```